core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala

```diff
@@ -474,6 +474,26 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with PrivateMethodTe
     assert(!BlockManagerId("notADriverIdentifier", "XXX", 1).isDriver)
   }
 
+  test("SPARK-43221: Host local block fetching should use a block status with disk size") {
+    conf.set(IO_ENCRYPTION_ENABLED, true)
```

Comment:
Is this required, @attilapiros ?

Reply:
Yes, it is needed for the reproduction of the error.
Otherwise we would read the complete file (use the file size instead of the size get from `getLocationsAndStatus` which might be the in-memory block size before this PR).

Size get from `BlockManagerMasterEndpoint#getLocationsAndStatus()` and passed as `blockSize`:
https://github.com/apache/spark/blob/4231d58245251a34ae80a38ea4bbf7d720caa439/core/src/main/scala/org/apache/spark/storage/BlockManager.scala#L1240

Filesize get as `file.length`:
https://github.com/apache/spark/blob/4231d58245251a34ae80a38ea4bbf7d720caa439/core/src/main/scala/org/apache/spark/storage/BlockManager.scala#L1244

Reply:
gentle ping @dongjoon-hyun